backend/src/main/controller/UserController.ts

```diff
@@ -0,0 +1,185 @@
+        try {
+            // get the id of user from the url
+            const id = await IdValidation.schema.validateAsync(req.params.id, {
+                abortEarly: false,
+            });
+            const userRepository = getCustomRepository(UserRepository);
+            const savedpromotions = await userRepository.findOneOrFail(id, {
+                relations: ["savedPromotions"]
+            });
+            
+            res.status(200).send(savedpromotions);
+        } catch (e) {
+            next(e);
+        }
+    };
+    // saved new promotion
+    newSaved = async (req: Request, res: Response, next: NextFunction) => {
+        try {
+            // get the id of user and promotion from the url
+            const uid = await IdValidation.schema.validateAsync(req.params.id, {
+                abortEarly: false,
+            });
+            const pid = await IdValidation.schema.validateAsync(req.params.pid, {
+                abortEarly: false,
+            });
+            // create new saved promotion for the user based on promotion id
+            const promotion = await getCustomRepository(PromotionRepository).findOneOrFail(pid);
+            const user = await getCustomRepository(UserRepository).findOneOrFail(uid);
+            const result = await getCustomRepository(UserRepository).addSavedPromotion(user, promotion);
+
+            res.status(201).send(result);
+        } catch (e) {
+            next(e);
+        }
+    };
+
+    // delete promotion
+    deleteSaved = async (req: Request, res: Response, next: NextFunction) => {
+        try {
+            // get the id of user and promotion from the url
+            const uid = await IdValidation.schema.validateAsync(req.params.id, {
+                abortEarly: false,
+            });
+            const pid = await IdValidation.schema.validateAsync(req.params.pid, {
+                abortEarly: false,
+            });
+            // get saved promotion and user
+            const promotion = await getCustomRepository(PromotionRepository).findOneOrFail(pid);
+            const user = await getCustomRepository(UserRepository).findOneOrFail(uid);
+            // remove the promotion from user's list of saved promotions
+            const result = await getCustomRepository(UserRepository).removeSavedPromotion(user, promotion);
+
+            return res.status(204).send(result);
+        } catch (e) {
+            next(e);
+        }
+    };
+
+    // get uploaded promotion
+    getUploaded = async (req: Request, res: Response, next: NextFunction) => {
+        try {
+            // get the id of user from the url
+            const id = await IdValidation.schema.validateAsync(req.params.id, {
+                abortEarly: false,
+            });
+            const userRepository = getCustomRepository(UserRepository);
+            const uploadedpromotions = await userRepository.findOneOrFail(id, {
+                relations: ["uploadedPromotions"]
+            });
```

Comment:
@andrewzulaybar  same for this part, how much detail do we want to show for uploaded promotions.

I was thinking the same reasoning for saved promotions.

I can imagine in the future we allow users to edit their uploaded promotions. Then when they click `edit` on their promotion in the `uploaded promotions page`, then we send a request to get all the necessary details and joins.